File: backend/src/app/Models/Broker.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use App\Traits\Auditable;

class Broker extends Model
{
    use HasFactory, SoftDeletes, Auditable;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'first_name',
        'last_name',
        'email',
        'phone',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
        'deleted_at' => 'datetime',
    ];

    /**
     * The accessors to append to the model's array form.
     *
     * @var array
     */
    protected $appends = ['full_name'];

    /**
     * Accessor: Full name
     */
    public function getFullNameAttribute(): string
    {
        return trim("{$this->first_name} {$this->last_name}");
    }

    /**
     * Relación: Un corredor puede estar asignado a un proveedor
     */
    public function provider()
    {
        return $this->hasOne(Provider::class);
    }

    /**
     * Override getAuditName to use full_name for Broker.
     */
    protected function getAuditName(): ?string
    {
        return $this->full_name;
    }
}

File: backend-dotnet/src/Ikhana.Domain/Entities/Broker.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Ikhana.Domain.Entities;

public class Broker : BaseEntity, ISoftDelete, IAuditable
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public DateTime? DeletedAt { get; set; }

    public long? ProviderId { get; set; }
    public Provider? Provider { get; set; }

    [NotMapped]
    public string FullName => $"{FirstName} {LastName}".Trim();
}
